### PR TITLE
Adding commands to quick installation

### PR DESCRIPTION
**Pull request by Aditya-eddy**
Gives feature mention in [issue](https://github.com/keploy/keploy/issues/1676)

### CODE REVIEW

**Comment by Aditya-eddy** [comment]
Hey @nehagup I have designed this for the website. Also thinking of adding different os in Available on section. WIll be waiting on feedback thanks!!
Screenshot from 2024-03-12 12-26-15

**Review by nehagup — CHANGES_REQUESTED** [review]
@Aditya-eddy thanks for the PR. This requires some changes:

- The font type and color needs to match the color theme of the website, please refer the readme. 
- The cli commands also needs to match the theme, please check the other cli-command in the features section. 
- this section needs to move above the "Join the Keploy community" section. 
- You can remove the box around the section and replace the "available on" with "Start Recording Test cases with code-less integration"
- The first command can just be `curl -O https://raw.githubusercontent.com/keploy/keploy/main/keploy.sh && source` 
- second command can be `keploy record -c 'go run main.go'`
- third command can be `keploy test -c 'go run main.go' --delay 10`
- On clicking the copy-button a "success message" snackbar can be displayed
- Please add "explore more on docs" with link under the last step.
- The first step command is breaking with responsiveness

**Comment by Aditya-eddy** [comment]
Hey @nehagup I have made the changes stated above. 
I have also  changed the color-theme. 
I have also fixed this [issue](https://github.com/keploy/keploy/issues/1681) in it.
Screenshot from 2024-03-15 04-08-50

**Comment by Aditya-eddy** [comment]
Sorry for the above mess. It has been fixed now. @nehagup

**Review by Hermione2408 — CHANGES_REQUESTED** [review]
@Aditya-eddy 

- As mentioned above this section should move above 'Community' section
- On clicking the copy-button a "success message" **snackbar** can be displayed (you can use from mui)
- Explore more on docs - should look clickable you can add a button similar to 'Documentation'
- Heading and subheading styling should be like 'Keploy for developers' section

Also please sign off DCO

**Comment by Aditya-eddy** [comment]
I have made these changes. Does they fullfill the above requests?
Screenshot from 2024-03-18 09-52-38

cc - @nehagup , @Hermione2408

File: app/layout.tsx [hunk 1]
```diff
@@ -4,7 +4,7 @@ import {Inter} from 'next/font/google'
 
 import Header from '@/components/ui/header'
 import React from 'react';
-
+import { AppRouterCacheProvider } from '@mui/material-nextjs/v13-appRouter';
```
Comment (Hermione2408) [inline app/layout.tsx]: Hey @Aditya-eddy I don't think it is required can't you directly use mui component in your section?

Comment (Aditya-eddy) [reply app/layout.tsx]: fixed it!

**Review by Hermione2408 — APPROVED** [review]
LGTM @nehagup

File: components/installation.tsx [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+import Link from "next/link";
+import Button from "@mui/material/Button";
+import Snackbar from "@mui/material/Snackbar";
+import IconButton from "@mui/material/IconButton";
+const Installation = () => {
+  const [checkedIndex, setCheckedIndex] = useState<number | null>(null);
+  const [open, setOpen] = React.useState(false);
+
+  const handleClick = () => {
+
```
Comment (nehagup) [inline components/installation.tsx]: Why is this empty?

File: components/installation.tsx [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+        </div>
+        <div className="flex justify-center">
+          <a
+            className="btn text-secondary-300 bg-primary-300 hover:font-semibold w-full mb-4 sm:w-auto sm:mb-0"
+            href="https://keploy.io/docs"
+          >
+            Explore more on Docs{" "}
+          </a>
+        </div>
+        <Snackbar
```
Comment (nehagup) [inline components/installation.tsx]: Can we make the styling same as the cli styling? will make it consistent

File: components/installation.tsx [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+          open={open}
+          autoHideDuration={6000}
```
Comment (nehagup) [inline components/installation.tsx]: 3 sec is also fine

File: components/installation.tsx [hunk 5]
```diff
@@ -0,0 +1,191 @@
...
+    if (reason === "clickaway") {
+      return;
+    }
+
+    setOpen(false);
+  };
+  const action = (
+    <React.Fragment>
+      <Button color="success" size="small" onClick={handleClose}>
+        UNDO
```
Comment (nehagup) [inline components/installation.tsx]: Instead of Undo we can add a close svg icon.

**Review by nehagup — COMMENTED** [review]
@aditya-eddy Thanks for the changes! A few more changes required to make it production ready. 

Screenshot 2024-03-19 at 2 53 02 PM

1. Copy button for Installing command is hidden for sm screen size. 
2. Can you please change the copy button to be same as one used in the one-click CLI command in the above section?
3. To increase the content in the section, we can add the following one-liners under each step? Please refer to the attached screenshot for the Step title, one liner. Also I've switched back to the CMD_TO_RUN_APP because we've added the description to explain this now. Please adjust the paddings and styling of code commands. I've also changed the button text for more relevancy. 
  
Screenshot 2024-03-19 at 4 10 27 PM

File: components/installation.tsx (hunk 2 above)
Comment (Aditya-eddy) [reply components/installation.tsx]: I've fixed it now!

File: components/installation.tsx (hunk 3 above)
Comment (Aditya-eddy) [reply components/installation.tsx]: Done!

File: components/installation.tsx (hunk 4 above)
Comment (Aditya-eddy) [reply components/installation.tsx]: OKay!!

File: components/installation.tsx (hunk 5 above)
Comment (Aditya-eddy) [reply components/installation.tsx]: Okay got it!

**Comment by Aditya-eddy** [comment]
Hey @nehagup how does this look or should i add some more padding?
If this looks good I'll push the commit.
Screenshot from 2024-03-20 04-25-03

Screenshot from 2024-03-20 04-26-22

**Review by Hermione2408 — CHANGES_REQUESTED** [review]
Hey @Aditya-eddy 
Screenshot from 2024-03-27 13-11-47
In the mobile view Snackbar is getting overlapped by the live chat icon , adjust the positioning to prevent this overlap. Also ensure the text on the button is centered.
Please remove the SVG that is not getting used now

File: components/installation.tsx [hunk 6]
```diff
@@ -0,0 +1,196 @@
...
+                src={checkedIndex === 1 ? "/images/tick.svg" : CopyButton}
+                height={25}
+                width={25}
+                onClick={() => handleclick(1)}
+                alt="Copy Icon"
+                className="cursor-pointer ml-2"
+              />
+            </div>
+          </div>
+            <p className=" text-gray-600 mb-0 mt-2 text-xs md:text-sm lg:text-base">
```
Comment (Hermione2408) [inline components/installation.tsx]: keep the text-color the same as other one-liner

**Review by Hermione2408 — CHANGES_REQUESTED** [review]

**Review by nehagup — COMMENTED** [review]
@Aditya-eddy good improvement. 

- Like @Hermione2408 mentioned please shorten the width of the snackbar so that it doesn't overlap. 
- Please add code formatting without quotes to the file names or cli commands, example `CMD_TO_RUN_APP` is the command to start you app. Say, `python main.py` or `npm start` or ...
- Also on smaller screen size can you please adjust the button to fix in one liner?

**Comment by Aditya-eddy** [comment]
Hey @nehagup how does this look. Here in the code blocks should give the a lighter theme or should give them same code blocks background as in the above code blocks
Screenshot from 2024-03-27 17-31-03
Screenshot from 2024-03-27 17-30-46

**Comment by nehagup** [comment]
> Hey @nehagup how does this look. Here in the code blocks should give the a lighter theme or should give them same code blocks background as in the above code blocks

@Aditya-eddy I think it's a good idea to keep it in lighter theme, so that it blends with the background. you can try to find a shade darker than bg-color of this page in tailwind config.

**Comment by Aditya-eddy** [comment]
Hey  @nehagup  does this look good:-
Screenshot from 2024-03-28 08-43-49

**Comment by Aditya-eddy** [comment]
Hey @nehagup any new changes in this?

**Comment by nehagup** [comment]
@Hermione2408 can you please review?

**Comment by Hermione2408** [comment]
Hey @Aditya-eddy there are conflicts in your PR please resolve them.  Also, it seems like the Installation component might have been missed in app/page.tsx during the conflict resolution. Could you check that?

**Comment by Hermione2408** [comment]
> @Hermione2408 can you please review?

Sure

**Comment by Aditya-eddy** [comment]
Yeah I forgot to add after resolving conflicts fixed it now. Check please! 
cc - @Hermione2408

File: components/installation.tsx [hunk 7]
```diff
@@ -0,0 +1,213 @@
...
+            Start Recording Test cases with code-less integration
+            <br />
+          </h1>
+        </div>
+
+        <div
+          className="max-w-4xl mx-auto pb-12 md:pb-16 "
+          data-aos="zoom-y-out"
+        >
+          <h1 className="text-2xl text-secondary-300 font-semibold leading-tighter tracking-tighter mb-2 ">
```
Comment (Hermione2408) [inline components/installation.tsx]: the margin is inconsistent please change it to mb-4

**Review by Hermione2408 — CHANGES_REQUESTED** [review]
A minor change from my side just resolve conflicts, rest looks fine!

File: components/installation.tsx (hunk 7 above)
Comment (Aditya-eddy) [reply components/installation.tsx]: hey resolved it!.

**Review by Hermione2408 — COMMENTED** [review]
LGTM @nehagup

**Review by nehagup — APPROVED** [review]
Thanks @Aditya-eddy for the PR! LGTM! 

There are spaces missing in Database,Redis,Kafka, but you can take that up in future PRs. :)

**Comment by Aditya-eddy** [comment]
I could have fixed in this only :smile:

**Pull request merged**